cardinal/ecs/component.go

```diff
@@ -213,3 +214,110 @@ func WithDefault[T any](defaultVal T) ComponentOption[T] {
 		c.validateDefaultVal()
 	}
 }
+
+// Get returns component data from the entity.
+func GetComponent[T component.INameable](w *World, id entity.ID) (comp *T, err error) {
+	var t T
+	name := t.Name()
+	c, ok := w.nameToComponent[name]
+	if !ok {
+		return nil, errors.New("Must register component")
+	}
+	value, err := w.StoreManager().GetComponentForEntity(c, id)
+	if err != nil {
+		return nil, err
+	}
+	t, ok = value.(T)
+	if !ok {
+		return nil, fmt.Errorf("type assertion for component failed: %v to %v", value, c)
+	}
+	comp = &t
+
+	return comp, nil
+}
+
+// Set sets component data to the entity.
+func SetComponent[T component.INameable](w *World, id entity.ID, component *T) error {
+	var t T
+	name := t.Name()
+	c, ok := w.nameToComponent[name]
+	if !ok {
+		return fmt.Errorf("%s is not registered, please register it before updating", t.Name())
+	}
+	err := w.StoreManager().SetComponentForEntity(c, id, component)
+	if err != nil {
+		return err
+	}
+	w.Logger.Debug().
+		Str("entity_id", strconv.FormatUint(uint64(id), 10)).
+		Str("component_name", c.Name()).
+		Int("component_id", int(c.ID())).
+		Msg("entity updated")
+	return nil
+}
+
+func UpdateComponent[T component.INameable](w *World, id entity.ID, fn func(*T) *T) error {
+	var t T
+	name := t.Name()
+	c, ok := w.nameToComponent[name]
+	if !ok {
+		return fmt.Errorf("%s is not registered, please register it before updating", t.Name())
+	}
+	if _, ok := w.nameToComponent[c.Name()]; !ok {
+		return fmt.Errorf("%s is not registered, please register it before updating", c.Name())
+	}
+	val, err := GetComponent[T](w, id)
+	if err != nil {
+		return err
+	}
+	updatedVal := fn(val)
+	return SetComponent[T](w, id, updatedVal)
+}
+
+//// EachComponent iterates over the entities that have the component.
+//// If you would like to stop the iteration, return false to the callback. To continue iterating, return true.
+//func EachComponent[T component.INameable](w *World, callback QueryCallBackFn) {
+//	var t T
+//	name := t.Name()
+//	c, ok := w.nameToComponent[name]
+//	if !ok {
+//		panic(fmt.Errorf("%s is not registered, please register it before updating", t.Name()))
+//	}
+//	c.GetQuery().Each(w, callback)
+//}
```

Comment:
should we just remove the component query stuff and require users to ONLY use the ecs.Query? this is esentially the same as `ecs.NewQuery(filter.Exact(FooComp).Each(....)`

Reply:
I'm cool with this. Yeah let's do that. I'll get rid of these functions unless anyone says otherwise.

Reply:
agreements? @namrapatel @jerargus @smsunarto

Reply:
@technicallyty i'll still do the package stuff in a separate PR. Keeps potential problems segregated and the PRs are easier to read.

Reply:
Yeah, a single blessed way to do queries sounds nice.